### PR TITLE
Solution to Egg-free DAG and Lazer Everywhere Alternative Problem in C++

### PR DESCRIPTION
**Pull request by Kuljeet-123**
**Fixes issue:** #4639  #4638 

**Changes:**

added solution to [EGGFREE](https://www.codechef.com/MARCH20A/problems/EGGFREE)
and [LAZERTST](https://www.codechef.com/MARCH20A/problems/LAZERTST) in C++.

### CODE REVIEW

File: code/online_challenges/src/codechef/BACREP/BACREP.cpp [hunk 1]
```diff
@@ -0,0 +1,111 @@
+#include<bits/stdc++.h>
```
Comment (arnavb) [inline code/online_challenges/src/codechef/BACREP/BACREP.cpp]: #1915

File: code/online_challenges/src/codechef/BACREP/BACREP.cpp [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+using namespace std;
```
Comment (arnavb) [inline code/online_challenges/src/codechef/BACREP/BACREP.cpp]: Don't use this.

File: code/online_challenges/src/codechef/BACREP/BACREP.cpp [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+#define ll long long
+#define dd double
+#define endl "\n"
+#define pb push_back
```
Comment (arnavb) [inline code/online_challenges/src/codechef/BACREP/BACREP.cpp]: Please remove all defines that you don't use.

File: code/online_challenges/src/codechef/BACREP/BACREP.cpp [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+ll mod=1000000007;
+const ll  N=500050;
+bool vis[N];
+vector <int> adj[N];
+ll depth[N]={0};
+vector <int> atd[N];
+ll maxdepth=1;
+ll par[N]={0};
+
+ll po(ll k ,ll n,ll ans,ll temp,ll ans1){
```
Comment (arnavb) [inline code/online_challenges/src/codechef/BACREP/BACREP.cpp]: Reformat your code using clang-format.

File: code/online_challenges/src/codechef/BACREP/BACREP.cpp [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    if(n==0)return ans;
+    while(temp<=n){
+        ans*=ans1;ans%=mod;ans1=ans1*ans1;ans1%=mod;n=n-temp;temp*=2;
```
Comment (arnavb) [inline code/online_challenges/src/codechef/BACREP/BACREP.cpp]: Separate lines.

File: code/online_challenges/src/codechef/HILLJUMP/HILLJUMP.cpp [hunk 6]
```diff
@@ -0,0 +1,44 @@
+#include<iostream>
```
Comment (arnavb) [inline code/online_challenges/src/codechef/HILLJUMP/HILLJUMP.cpp]: Same issues as your other file.

**Comment by Kuljeet-123** [comment]
@arnavb I have updated all the code and files.

**Comment by Kuljeet-123** [comment]
@arnavb Please review it and merge it also.

**Comment by arnavb** [comment]
Separate your PRs and implement my comments.

**Comment by Kuljeet-123** [comment]
@arnavb Due to conflicts , I am closing this PR and will raise a new PR. Because I  forked the repo again

**Pull request closed without merging**